functions/modelPreparation.R

```diff
@@ -46,6 +46,26 @@ modelPreparation <- function(focalTaxa, speciesData, redListModelled, regionGeom
     })
     focalSpeciesDataRefined <- focalSpeciesDataRefined[!is.na(focalSpeciesDataRefined)]
     
+    
+    # Combine species by functionalGroup if requested (else leave as separate)
+    # identify species with data
+    uniqueTaxaSpecies <- unique(unlist(lapply(focalSpeciesDataRefined, function(ds){
+      as.character(ds$acceptedScientificName)
+    })))
+    
+    # identify functional groups in species with data for focal taxonomic group
+    focalSpeciesWithData <- focalTaxa[focalTaxa$key %in% uniqueTaxaSpecies &  # species with data
+                                        focalTaxa$taxa %in% focalTaxaGroup,]  # and of focal taxa (in case same species in different taxa)
+    # if any species are to be modelled as functional groups
+    if(any(focalSpeciesWithData$functionalGroup != "")){
+      # update data
+      focalSpeciesDataRefined <- joinFunctionalGroups(speciesData = focalSpeciesDataRefined,
+                                                         focalTaxon = focalSpeciesWithData) 
+      focalGroupSpecies <- unique(unlist(lapply(focalSpeciesDataRefined, function(ds){
+        as.character(ds$acceptedScientificName)
+      })))
+    }
+    
     if (length(focalSpeciesDataRefined) == 0) {
```

Comment:
I think this bit should be duplicated and added to what is currently line 49, in case we have no data much earlier